### PR TITLE
arm: handle v5 mode

### PR DESCRIPTION
**Pull request by aohbrien**
Adds ARMv5 support.

### CODE REVIEW

**Comment by aquynh** [comment]
cool, but can you please fix the bugs found by CI?

https://travis-ci.org/keystone-engine/keystone/jobs/259379126

**Comment by aquynh** [comment]
looks good now. for the record, can you provide some sample armv6 assembly code for testing?

File: suite/regress/arm_v5.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# Author: Andrew O'Brien
+
+from keystone import (Ks, KsError, KS_ARCH_ARM, KS_MODE_ARM, KS_MODE_V5)
+import regress
+
+class TestARM(regress.RegressTest):
+    asm = b"""
+    MOVT r0, #0
+    """
```
Comment (aquynh) [inline suite/regress/arm_v5.py]: this sample code can also assembled by armv7 i think, with the same result. is there any code that is only available in armv5?

Comment (aohbrien) [inline suite/regress/arm_v5.py]: I don't think there are any ARMv5 instructions that don't also exist in newer versions, for backwards compatibility. Only deprecation warnings are issued in the migration manuals in favor of newer features, but no breaking changes are made.

**Comment by aquynh** [comment]
Cool, then when is this new v5 mode useful?

**Pull request closed without merging**